projects/gloo/pkg/plugins/istio_automtls/helper.go

```diff
@@ -0,0 +1,35 @@
+package istio_automtls
+
+import (
+	"github.com/solo-io/gloo/projects/gloo/constants"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+)
+
+const EnvoyTransportSocketMatch = "envoy.transport_socket_match"
+
+// AddIstioAutomtlsMetadata adds metadata used by the transport_socket_match
+// to select the mTLS transport socket. The Envoy metadata label is added
+// based on the presence of the Istio workload label "security.istio.io/tlsMode=istio".
+func AddIstioAutomtlsMetadata(
+	metadata *envoy_config_core_v3.Metadata,
+	workloadLabels map[string]string,
+	enableAutoMtls bool,
+) *envoy_config_core_v3.Metadata {
+	if enableAutoMtls {
+		// Valid label values are 'istio', 'disabled'
```

Comment:
is it possible to add a link to the istio ref for these values? I already have forgotten the istio semantics regarding these

Reply:
also just for for clarity i was thinking adding this link to the comment

Reply:
Linked to the Istio API def that outlines it.